scripts/secnetperf.ps1

```diff
@@ -184,13 +184,27 @@ VALUES ("$MsQuicCommit", "$(Get-Date -Format "yyyy-MM-dd HH:mm:ss")", 1, "TODO")
 "@
 $json = @{}
 $json["commit"] = "$MsQuicCommit"
+# Persist environment information:
+if ($isWindows) {
+    $windowsEnv = Get-CimInstance Win32_OperatingSystem | Select-Object Version
+    $json["os_version"] = $windowsEnv.Version
+} else {
+    $osInfo = bash -c "cat /etc/os-release"
+    $osInfoLines = $osInfo -split "`n"
+    $osName = $osInfoLines | Where-Object { $_ -match '^PRETTY_NAME=' } | ForEach-Object { $_ -replace '^PRETTY_NAME="|"$', '' }
+    $kernelVersion = bash -c "uname -r"
+    $json["os_version"] = "$osName $kernelVersion"
+}
 $allTests = [System.Collections.Specialized.OrderedDictionary]::new()
 
+# Create file to persist full latency curve data:
+New-Item -ItemType File -Name "full-latency-curve.txt"
+
 # > All tests:
 $allTests["tput-up"] = "-exec:maxtput -up:12s -ptput:1"
 $allTests["tput-down"] = "-exec:maxtput -down:12s -ptput:1"
 $allTests["hps-conns-100"] = "-exec:maxtput -rconn:1 -share:1 -conns:100 -run:12s -prate:1"
-$allTests["rps-up-512-down-4000"] = "-exec:lowlat -rstream:1 -up:512 -down:4000 -run:20s -plat:1"
+$allTests["rps-up-512-down-4000"] = "-exec:lowlat -rstream:1 -up:512 -down:4000 -run:20s -plat:1 -extraOutputFile:full-latency-curve.txt"
```

Comment:
1. I think we should fully specify path.
2. Let's not put this here, but instead, whenever we execute an RPS test (in the helper script) always append this argument to the client (not server).
3. Let's name this file '<test-name>-latency.txt`.
4. We are going to want to post-process this file to a nicer format for automation to use.  We can either do it here, or later. IMHO, doing it here, so we only ever upload the nicer format is the best option.

Reply:
For post-processing to a more condensed format, my thoughts were to merge all the latency.txt files produced from all the separate environment runs into 1 file, and commit that to some orphan branch with the commit name. This step would happen after all the runs for the different environments have completed. Thoughts?

Reply:
I'm fine with merging into a single file, but I don't want to go creating tons or branches. I'd rather a folder with the commit name in a single branch.